internal/provider/resource_gitlab_group_saml_link.go

```diff
@@ -0,0 +1,163 @@
+				Description: "The name of the SAML group.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"force": {
+				Description: "If true, then delete and replace an existing SAML link if one exists.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+			},
+		},
+	}
+})
+
+func resourceGitlabGroupSamlLinkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupId := d.Get("group_id").(string)
+	accessLevel := d.Get("access_level").(string)
+	samlGroupName := d.Get("saml_group_name").(string)
+	force := d.Get("force").(bool)
+
+	options := &gitlab.AddGroupSAMLLinkOptions{
+		AccessLevel:   &accessLevel,
+		SamlGroupName: &samlGroupName,
+	}
+
+	if force {
+		if err := resourceGitlabGroupSamlLinkDelete(ctx, d, meta); err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] Create GitLab group SamlLink %s", d.Id())
+	SamlLink, _, err := client.Groups.AddGroupSAMLLink(groupId, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(buildTwoPartID(&groupId, &SamlLink.Name))
+
+	return resourceGitlabGroupSamlLinkRead(ctx, d, meta)
+}
+
+func resourceGitlabGroupSamlLinkRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	groupId := d.Get("group_id").(string)
+
+	// Try to fetch all group links from GitLab
+	log.Printf("[DEBUG] Read GitLab group SamlLinks %s", groupId)
+	samlLinks, _, err := client.Groups.ListGroupSAMLLinks(groupId, nil, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// If we got here and don't have links, assume GitLab is below version 12.8 and skip the check
+	if samlLinks != nil {
+		// Check if the LDAP link exists in the returned list of links
+		found := false
+		for _, samlLink := range samlLinks {
+			if buildTwoPartID(&groupId, &samlLink.Name) == d.Id() {
+				d.Set("group_id", groupId)
+				d.Set("access_level", samlLink.AccessLevel)
+				d.Set("saml_group_name", samlLink.Name)
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			d.SetId("")
+			return diag.Errorf("SamlLink %s does not exist.", d.Id())
+		}
+	}
+
+	return nil
+}
+
+func resourceGitlabGroupSamlLinkDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	groupId := d.Get("group_id").(string)
+	samlGroupName := d.Get("saml_group_name").(string)
+
+	log.Printf("[DEBUG] Delete GitLab group SamlLink %s", d.Id())
+	_, err := client.Groups.DeleteGroupSAMLLink(groupId, samlGroupName, cn, gitlab.WithContext(ctx))
+	if err != nil {
+		switch err.(type) { // nolint // TODO: Resolve this golangci-lint issue: S1034: assigning the result of this type assertion to a variable (switch err := err.(type)) could eliminate type assertions in switch cases (gosimple)
```

Comment:
Not sure about the error parsing here and the use of `// nolint` ... maybe you can use something similar to this instead:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/630a88e620f604322d8e28cc8ed7f16558ecb858/internal/provider/variable_helpers.go#L14